lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -928,6 +928,9 @@ unsigned SharedEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
 SmallVector<unsigned>
 DotOperandEncodingAttr::getElemsPerThread(ArrayRef<int64_t> shape,
                                           Type eltTy) const {
+  if (auto mmaParent = mlir::dyn_cast<MmaEncodingTrait>(getParent())) {
+    return mmaParent.getElemsPerThreadForOperands(shape, eltTy, getOpIdx());
+  }
```

Comment:
I'm not sure this is always correct? The MMA layouts don't always match the associated dot operand layout?